sapi/phpdbg/phpdbg_frame.c

```diff
@@ -269,42 +267,45 @@ void phpdbg_dump_backtrace(size_t num) /* {{{ */
 		return;
 	} phpdbg_end_try_access();
 
-	Z_LVAL(startline) = zend_get_executed_lineno();
-	startfilename = zend_get_executed_filename();
-	Z_STR(startfile) = zend_string_init(startfilename, strlen(startfilename), 0);
-
-	zend_hash_internal_pointer_reset_ex(Z_ARRVAL(zbacktrace), &position);
+	line = zend_get_executed_lineno();
+	file = zend_get_executed_filename_ex();
 
 	zval *function_name = NULL;
-	while ((tmp = zend_hash_get_current_data_ex(Z_ARRVAL(zbacktrace), &position))) {
+	ZEND_HASH_FOREACH_VAL(Z_ARRVAL(zbacktrace), tmp) {
 		if (file) { /* userland */
 			phpdbg_out("frame #%d: ", i);
 			phpdbg_dump_prototype(tmp);
-			phpdbg_out(" at %s:"ZEND_LONG_FMT"\n", Z_STRVAL_P(file), Z_LVAL_P(line));
+			phpdbg_out(" at %s:"ZEND_LONG_FMT"\n", ZSTR_VAL(file), line);
 			i++;
 		} else {
 			phpdbg_out(" => ");
 			phpdbg_dump_prototype(tmp);
 			phpdbg_out(" (internal function)\n");
 		}
 
-		file = zend_hash_find(Z_ARRVAL_P(tmp), ZSTR_KNOWN(ZEND_STR_FILE));
-		line = zend_hash_find(Z_ARRVAL_P(tmp), ZSTR_KNOWN(ZEND_STR_LINE));
+		zval *file_zv = zend_hash_find(Z_ARRVAL_P(tmp), ZSTR_KNOWN(ZEND_STR_FILE));
```

Comment:
Just change this to
`file = zend_hash_find_ptr(Z_ARRVAL_P(tmp), ZSTR_KNOWN(ZEND_STR_FILE));`
and remove the if-else part.
That makes the code less noisy.